extensions/amp-social-share/1.0/social-share.js

```diff
@@ -57,6 +58,17 @@ export function SocialShare(props) {
     width: checkedWidth,
     height: checkedHeight,
   };
+
+  const icon =
+    !getSocialConfig(props['type']) || props['customIcon'] ? (
```

Comment:
Is `customIcon` needed? Can you do `props['children'] || <SocialShareIcon ...>`?

Reply:
From my understanding `props['children']` creates an unnamed slot in the shadow DOM (when using `passthrough`) so the children can be displayed, but actually does not represent those children.

And therefore from Preact side, there is actually no way of knowing whether or not the AMP side has provided children.

This is the basis for using `customIcon`, but if I'm not understanding correctly happy to be corrected!

Reply:
@krdwan is correct that currently, the "passthrough" will pass the content as a single `slot` no matter whether or not there are children. But @caroqliu is correct that burdening the component with this nuance seems wrong. I think we just need to make `passthrough` smarter. The simplest thing we could for now is to create a "passthrough-non-empty" mode where we create a shadow DOM, but compute children as following:

```
props['children'] = this.getRealChildNodes().length > 0 ? [<Slot />] : null;
```